### PR TITLE
Fix for #5 Use index in ExampleNameProvider

### PR DESCRIPTION
**Pull request by ghaith**
Added an index method to fetch the name
removed NonStoringBatchLinkableResourceStorageFacade

### CODE REVIEW

**Comment by riederm** [comment]
were you able to reproduce the issue first?

**Comment by ghaith** [comment]
Yes, disable the non storing facade, create a spec and make a suite that
references it, or references all, you will get a null pointer exception
I'm not sure if this is the best fix for it though, I kept the extra
nodemodel access as backup in it

File: plugins/org.jnario.spec/src/org/jnario/spec/scoping/SpecResourceDescriptionStrategy.xtend [hunk 1]
```diff
@@ -13,18 +13,27 @@ import org.jnario.scoping.JnarioResourceDescriptionStrategy
...
 	
 	override createUserData(EObject eObject, ImmutableMap.Builder<String, String> userData) {
 		super.createUserData(eObject, userData);
 		if (eObject instanceof ExampleGroup){
 			val exampleGroup = eObject as ExampleGroup			
 			userData.put(ROOT_SPEC, valueOf(exampleGroup.isRoot))
+			if (exampleGroup.targetType !== null) {	
+				val targetType = exampleGroup.targetType
+				if (!(targetType.type instanceof JvmVoid)) {
+					userData.put(EXAMPLE_TYPE,exampleGroup.targetType.simpleName)
```
Comment (riederm) [inline plugins/org.jnario.spec/src/org/jnario/spec/scoping/SpecResourceDescriptionStrategy.xtend]: I'm not sure if we can safely call exampleGroup.getTargetType(). This has to sometimes return an unresolvable proxy? Otherwhise the original problem of the ExampleNameProvider wouldn't exist. 

I think the problem is that you write a spec:

MySpec.spec
```
describe MyClass{
}
```

MyClass.xtend
```
class MyClass{
}
```
So the ExampleGroup in MySpec.spec holds a reference targetType = MyClass,
I think the unresolvable problem happens, if MyClass was not generated yet, while you're indexing. E.g. in a cleanBuild.

I would rather move the NodeModel-approach to here, and remove it from the ExampleNameProvider. 
If we really want to make sure, we should also store the nodemodel in the storage (although it means a lot of more stuff)

I just checked, by default the node-model is not stored in the storage, but you can tell the storage facade to store the node-model as well. But its again a performance loss :( 
org.eclipse.xtext.resource.persistence.ResourceStorageFacade.setStoreNodeModel(boolean)

**Comment by ghaith** [comment]
I uploaded a second attempt, i cannot reproduce an error where the class has not been generated yet, i kept fallbacks for the nodemodel but moved the requests to the spec description factory
The original error, where a suite will get a nullpointer when referencing the non resolved spec is also not occuring anymore. Still not sure if we should keep the fallbacks or if you had any other way to test this

**Review by riederm — APPROVED** [review]

**Pull request merged**